lib/src/layer/tile_layer/tile_layer.dart

```diff
@@ -584,7 +591,19 @@ class _TileLayerState extends State<TileLayer> with TickerProviderStateMixin {
     center ??= map.center;
 
     final pixelBounds = _getTiledPixelBounds(map, center);
-    final tileRange = _pxBoundsToTileRange(pixelBounds);
+    Bounds tileRange = _pxBoundsToTileRange(pixelBounds);
+
+    final panBuffer = widget.panBuffer;
+
+    // Increase the tilerange if we have panBuffer set, but make sure we
+    // don't use values outside valid tiles, eg (0,-1).
+    tileRange = tileRange.extend(CustomPoint(
+        math.max(_globalTileRange.min.x, tileRange.min.x - panBuffer),
+        math.max(_globalTileRange.min.y, tileRange.min.y - panBuffer)));
+    tileRange = tileRange.extend(CustomPoint(
```

Comment:
We could use a condition `if (panBuffer > 0)` to avoid unecessary computation on `tileRange` 🤔 
Otherwise the code looks good to me

Reply:
Yes, I think I like that, every little helps!

Reply:
Ok, have added an if( panBuffer != 0), maybe it should be > 0, but maybe there is some weird reason for someone to reduce it as well that I can't think of (although maybe there could be bugs if they did that, but there is a bit of caveat emptor or whatever it is)!

Reply:
Was thinking about it too but I supposed that it wouldn't make sense to buffer `-1` tiles 🤔